tests/python_client/testcases/test_query.py

```diff
@@ -554,6 +554,131 @@ def test_query_expr_non_constant_array_term(self):
+                ct.default_float_vec_field_name: cf.gen_vectors(1, ct.default_dim)[0]
+            }
+            array.append(data)
+        collection_w.insert(array)
+
+        # 3. query
+        collection_w.load()
+        expressions = ["json_contains(json_field['list'], 1000)", "JSON_CONTAINS(json_field['list'], 1000)"]
+        for expression in expressions:
+            res = collection_w.query(expression)[0]
+            assert len(res) == limit
+
+    @pytest.mark.tags(CaseLabel.L2)
+    def test_query_expr_list_json_contains(self):
+        """
+        target: test query with expression using json_contains
+        method: query with expression using json_contains
+        expected: succeed
+        """
+        # 1. initialize with data
+        collection_w = self.init_collection_general(prefix, enable_dynamic_field=True)[0]
+
+        # 2. insert data
+        limit = ct.default_nb // 4
+        array = []
+        for i in range(ct.default_nb):
+            data = {
+                ct.default_int64_field_name: i,
+                ct.default_json_field_name: [str(m) for m in range(i, i + limit)],
+                ct.default_float_vec_field_name: cf.gen_vectors(1, ct.default_dim)[0]
+            }
+            array.append(data)
+        collection_w.insert(array)
+
+        # 3. query
+        collection_w.load()
+        expressions = ["json_contains(json_field, '1000')", "JSON_CONTAINS(json_field, '1000')"]
+        for expression in expressions:
+            res = collection_w.query(expression, output_fields=["count(*)"])[0]
+            assert len(res) == limit
+
+    @pytest.mark.tags(CaseLabel.L2)
+    def test_query_expr_json_contains_combined_with_normal(self, enable_dynamic_field):
+        """
+        target: test query with expression using json_contains
+        method: query with expression using json_contains
+        expected: succeed
+        """
+        # 1. initialize with data
+        collection_w = self.init_collection_general(prefix, enable_dynamic_field=enable_dynamic_field)[0]
+
+        # 2. insert data
+        limit = ct.default_nb // 3
+        array = []
+        for i in range(ct.default_nb):
+            data = {
+                ct.default_int64_field_name: i,
+                ct.default_float_field_name: i * 1.0,
+                ct.default_string_field_name: str(i),
+                ct.default_json_field_name: {"number": i, "list": [m for m in range(i, i + limit)]},
+                ct.default_float_vec_field_name: cf.gen_vectors(1, ct.default_dim)[0]
+            }
+            array.append(data)
+        collection_w.insert(array)
+
+        # 3. query
+        collection_w.load()
+        tar = 1000
+        expressions = [f"json_contains(json_field['list'], {tar}) && float > {tar - limit // 2}",
+                       f"JSON_CONTAINS(json_field['list'], {tar}) && float > {tar - limit // 2}"]
```

Comment:
json_field['list'] and json_field both ok in json_contains, how about json_contains('list', 1000) in both dynamic and non-dynamic field, should add these test cases in query and search.

Reply:
> json_field['list'] and json_field both ok in json_contains, how about json_contains('list', 1000) in both dynamic and non-dynamic field, should add these test cases in query and search.

add logic in test_search_with_expression_json_contains and test_query_expr_json_contains